File: app/src/main/java/com/chrisprime/primestationonecontrol/dagger/modules/ThreadPoolModule.java
package com.chrisprime.primestationonecontrol.dagger.modules;


import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ThreadPoolModule {

    public ThreadPoolModule() {

    }

    @Provides
    @Singleton
    ExecutorService provideExecutorService() {
        return Executors.newCachedThreadPool();
    }
}

File: app/src/main/java/com/chrisprime/primestationonecontrol/dagger/modules/ThreadPoolModule.kt
package com.chrisprime.primestationonecontrol.dagger.modules


import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
class ThreadPoolModule {

    @Provides
    @Singleton
    fun provideExecutorService(): ExecutorService {
        return Executors.newCachedThreadPool()
    }
}
